web3.js/src/transaction.ts

```diff
@@ -169,13 +202,41 @@ export class Transaction {
    */
   nonceInfo?: NonceInformation;
 
+  /**
+   * @internal
+   */
+  _message?: Message;
+
+  /**
+   * @internal
+   */
+  _json?: TransactionJSON;
+
   /**
    * Construct an empty Transaction
    */
   constructor(opts?: TransactionCtorFields) {
     opts && Object.assign(this, opts);
   }
 
+  toJSON(): TransactionJSON {
+    return {
+      recentBlockhash: this.recentBlockhash || null,
+      feePayer: this.feePayer ? this.feePayer.toJSON() : null,
+      nonceInfo: this.nonceInfo
+        ? {
+            nonce: this.nonceInfo.nonce,
+            nonceInstruction: this.nonceInfo.nonceInstruction.toJSON(),
+          }
+        : null,
+      instructions: this.instructions.map(instruction => instruction.toJSON()),
+      signatures: this.signatures.map(({publicKey, signature}) => ({
+        publicKey: publicKey.toJSON(),
```

Comment:
I don't love that it's `publicKey` here and `pubkey` in the instructions. Not sure if we should preserve consistency with the type or make for a predictable interface.

Reply:
I think the name discrepancy is ok, but can we make `toJSON` an internal method?

Reply:
Yeah, I can do that. Always nice to kick the can on changes to the public API.